classes/ObjectModel.php

```diff
@@ -385,12 +391,15 @@ public function getFieldsLang()
      */
     public function getAssociatedLanguage(): Language
     {
-        $language = new Language($this->id_lang);
-        if (null === $language->id) {
-            $language = new Language((int) Configuration::get('PS_LANG_DEFAULT'));
+        if (!is_null($this->lang_associated)) {
```

Comment:
Just to be peaky 😅:
```suggestion
        if (null !== $this->lang_associated) {
```

Reply:
@atomiix Yoda stile or another reason?

Reply:
using `===` instead of the function `is_null()`. Mainly because that's what you also did just after so it's coherent (but it's also a micro optimization as using `is_null()` is slightly slower).

Reply:
@atomiix Done.